### ISSUE TITLE
`Style/SymbolArray` warns when valid

### ISSUE THREAD

**Issue by SaladFork** [bbatsov/rubocop#5078]
## Expected behavior

`Style/SymbolArray` should not complain when using a symbol array.

## Actual behavior

`Style/SymbolArray` complains when using a symbol array.

## Steps to reproduce the problem

```ruby
# example.rb

class Example
  NUMBERS = %i[one two three]
end
```

```bash
$ rubocop --show-cops 'Style/SymbolArray'
# Supports --auto-correct
Style/SymbolArray:
  Description: Use %i or %I for arrays of symbols.
  StyleGuide: "#percent-i"
  Enabled: true
  EnforcedStyle: brackets
  MinSize: 0
  SupportedStyles:
  - percent
  - brackets
```

```bash
$ rubocop example.rb --only 'Style/SymbolArray' -D
Inspecting 1 file
C

Offenses:

example.rb:2:13: C: Style/SymbolArray: Use [] for an array of symbols.
  NUMBERS = %i[one two three]
            ^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.0 x86_64-darwin16)
```

**Comment by Drenmi** [comment]
You can see from the configuration snippet you provided that it has been configured to enforce bracket notation, not percent notation. 🙂 

> ```
> $ rubocop --show-cops 'Style/SymbolArray'
> # Supports --auto-correct
> Style/SymbolArray:
>   Description: Use %i or %I for arrays of symbols.
>   StyleGuide: "#percent-i"
>   Enabled: true
>   EnforcedStyle: **brackets**
>   MinSize: 0
>   SupportedStyles:
>   - percent
>   - brackets
> ```

**Comment by SaladFork** [comment]
🤦 wow, thank you... sorry for the noise.

**Comment by Drenmi** [comment]
No worries. Will have a think about whether this name is sufficient, since the symbol notation technically has brackets as well. 🤔

**Comment by SaladFork** [comment]
I was actually confusing this cop with `Style/PercentLiteralDelimiters`, which has a very similar fail message (`%i-literals should be delimited by [ and ]`).
